### PR TITLE
♻️ Better bound implementation

### PR DESCRIPTION
**Pull request by ZeroEkkusu**
## Description

Change the implementation of `bound` to perfectly bound the input every time:

### Logic

***`x` ----> `result`***

```
0 ------> `min`
1 ------> `min` + 1
2 ------> `min` + 2
...
n ------> `max`
n + 1 --> `min`
n + 2 --> `min` + 1
...
```

### Examples

You can find numerous examples [here](https://github.com/foundry-rs/forge-std/pull/57#pullrequestreview-958608080).

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge snapshot`?
- [x] Ran `npm run lint`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

**Comment by transmissions11** [comment]
great work! can you run `npm run lint`?

**Comment by ZeroEkkusu** [comment]
> great work! can you run `npm run lint`?

first time using it. i ran it and got this; didn't change anything.

```
00x@ubuntu:~/Documents/00x/solmate$ npm run lint

> @rari-capital/solmate@6.2.0 lint
> prettier --write src/**/*.sol

src/auth/Auth.sol 248ms
...
```

**Comment by transmissions11** [comment]
> first time using it. i ran it and got this; didn't change anything.


ah yup yup looks like the script wasn't setup properly, good find!

File: src/test/utils/DSTestPlus.sol [hunk 1]
```diff
@@ -116,21 +116,20 @@ contract DSTestPlus is DSTest {
         uint256 x,
         uint256 min,
         uint256 max
-    ) internal returns (uint256 result) {
-        require(max >= min, "MAX_LESS_THAN_MIN");
+    ) public returns (uint256 result) {
```
Comment (transmissions11) [inline src/test/utils/DSTestPlus.sol:119]: why public?

File: src/test/utils/DSTestPlus.sol [hunk 2]
```diff
@@ -116,21 +116,20 @@ contract DSTestPlus is DSTest {
...
+        require(min <= max, "MAX_LESS_THAN_MIN");
```
Comment (transmissions11) [inline src/test/utils/DSTestPlus.sol:120]: i prefer the previous ordering

File: src/test/utils/DSTestPlus.sol (hunk 1 above)
Comment (ZeroEkkusu) [reply src/test/utils/DSTestPlus.sol:119]: copy-pasted from forge std and forgot

**Comment by mds1** [comment]
Since afaik solmate now relies on forge, instead of upstreaming this why not just use the forge-std version directly? Similarly, in solmate's DSTestPlus we can remove `assertFalse`, `assertApproxEq`, and `assertRelApproxEq` and use the forge-std versions there too.

Right now you cannot have `contract MyDSTestPlus is SolmatesDSTestPlus, Test` because of duplicate `assertFalse` definitions in solmate and forge-std, so IMO we should remove duplication of methods between solmate and forge-std

**Comment by ZeroEkkusu** [comment]
Solmate will continue using DSTest until #214 get merged.

There are no conflicts atm.

**Comment by transmissions11** [comment]
great work thank you so much @ZeroEkkusu !!

**Comment by transmissions11** [comment]
@mds1 will remove bound once we switch to use forge-std, gotta finish reviewing andreas' PR

**Pull request merged**